src/storage/mod.rs

```diff
@@ -254,7 +254,7 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         mut ctx: Context,
         key: Key,
         start_ts: TimeStamp,
-    ) -> impl Future<Output = Result<Option<Value>>> {
+    ) -> impl Future<Output = Result<(Option<Value>, Statistics, PerfStatisticsDelta)>> {
```

Comment:
If we change the return type to `(Result<Option<Value>>, Statistics, PerfStatisticsDelta)`, function like `extract_kv_pairs_and_statistics` could be elimated then codes would be cleaner. How do you think?

Reply:
Sorry,  I don't see what you mean.

Reply:
`extract_kv_pairs` takes `Result<Vec<Result<KvPair>>>`, so we won't need to introduce `extract_kv_pairs_and_statistics` if we can extract `Result<Vec<Result<KvPair>>>` from `batch_get`'s return type easily.

`(Result<Vec<Result<KvPair>>>, Statistics, PerfStatisticsDelta)` is such a type that satisfies this purpose.

Reply:
Ok, I got it. Let me try.